docs/modules/ROOT/pages/components.adoc

```diff
@@ -344,6 +344,91 @@ mod MyContract {
+The following snippet leverages the `before_update` hook to include this behavior.
+
+[,cairo]
+----
+#[starknet::contract]
+mod MyToken {
+    use openzeppelin::security::pausable::PausableComponent::InternalTrait;
+    use openzeppelin::security::pausable::PausableComponent;
+    use openzeppelin::token::erc20::ERC20Component;
+    use starknet::ContractAddress;
+
+    component!(path: ERC20Component, storage: erc20, event: ERC20Event);
+    component!(path: PausableComponent, storage: pausable, event: PausableEvent);
+
+    // ERC20 Mixin
+    #[abi(embed_v0)]
+    impl ERC20MixinImpl = ERC20Component::ERC20MixinImpl<ContractState>;
+    impl ERC20InternalImpl = ERC20Component::InternalImpl<ContractState>;
+
+    #[abi(embed_v0)]
+    impl PausableImpl = PausableComponent::PausableImpl<ContractState>;
+    impl PausableInternalImpl = PausableComponent::InternalImpl<ContractState>;
+
+    // Create the hooks implementation
+    impl ERC20HooksImpl of ERC20Component::ERC20HooksTrait<ContractState> {
+        // Occurs before token transfers
+        fn before_update(
+            ref self: ERC20Component::ComponentState<ContractState>,
+            from: ContractAddress,
+            recipient: ContractAddress,
+            amount: u256
+        ) {
+            // Access local state from component state
+            let contract_state = ERC20Component::HasComponent::get_contract(@self);
+            // Call function from integrated component
+            contract_state.pausable.assert_not_paused();
+        }
+
+        // Occurs after token transfers
+        fn after_update(
+            ref self: ERC20Component::ComponentState<ContractState>,
+            from: ContractAddress,
+            recipient: ContractAddress,
+            amount: u256
+        ) {}
+    }
+
+    (...)
+}
+----
+
+Notice that the `self` parameter expects a component state type.
+Instead of passing the component state, the using contract's state can be passed which simplifies the syntax.
+The hook then moves the scope up with the Cairo-generated `get_contract` through the `HasComponent` trait (through ERC20Component in this example).
+From here, the hook can access the using contract's integrated components, storage, and implementations.
+
+Be advised that even if a token contract does not require hooks, the hooks trait must still be implemented.
+Contracts may implement the trait with empty functions.
+The Contracts for Cairo library, however, offers default empty hook implementations to abstract this away from contracts.
+Rather than defining the hooks implementation, the using contract just needs to bring the default implementation into scope like this:
+
+[,cairo]
+----
+#[starknet::contract]
+mod MyToken {
+    use openzeppelin::token::erc20::ERC20Component;
+    use openzeppelin::token::erc20::ERC20HooksEmptyImpl;
+
+    (...)
+}
+----
+
+See {hooks-post} for a more in-depth guide.
```

Comment:
Does it make sense to move the link to the post to the section's header?

Reply:
I don't think it'd be awful. What benefits do you see for the reader by moving it to the beginning?

Reply:
Hm I agree it'll be more visible, but I imagine it's better for the reader to read the section and then see the link at the end if they want more info (maybe?). I included a TIP block to better highlight it. Let me know what you think

Reply:
I think there's no big difference, the TIP approach works for me